<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Supplemental Tutorial | Supplementary Materials - CellDepot: A unified repository for scRNAseq data and visual exploration</title>
  <meta name="description" content="Chapter 4 Supplemental Tutorial | Supplementary Materials - CellDepot: A unified repository for scRNAseq data and visual exploration" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Supplemental Tutorial | Supplementary Materials - CellDepot: A unified repository for scRNAseq data and visual exploration" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="https://github.com/interactivereport/CellDepot" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Supplemental Tutorial | Supplementary Materials - CellDepot: A unified repository for scRNAseq data and visual exploration" />
  
  
  



<meta name="date" content="2021-12-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="SITable.html"/>

<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CellDepot</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="getting-start-with-celldepot.html"><a href="getting-start-with-celldepot.html"><i class="fa fa-check"></i><b>2</b> Getting start with CellDepot</a>
<ul>
<li class="chapter" data-level="2.1" data-path="getting-start-with-celldepot.html"><a href="getting-start-with-celldepot.html#sources-of-annotation-and-metadata"><i class="fa fa-check"></i><b>2.1</b> Sources of annotation and metadata</a></li>
<li class="chapter" data-level="2.2" data-path="getting-start-with-celldepot.html"><a href="getting-start-with-celldepot.html#data-format-availability-and-preparation"><i class="fa fa-check"></i><b>2.2</b> Data format, availability, and preparation</a></li>
<li class="chapter" data-level="2.3" data-path="getting-start-with-celldepot.html"><a href="getting-start-with-celldepot.html#celldepot-platform-and-installation"><i class="fa fa-check"></i><b>2.3</b> CellDepot platform and installation</a></li>
<li class="chapter" data-level="2.4" data-path="getting-start-with-celldepot.html"><a href="getting-start-with-celldepot.html#cron"><i class="fa fa-check"></i><b>2.4</b> How to set up cron job?</a></li>
<li class="chapter" data-level="2.5" data-path="getting-start-with-celldepot.html"><a href="getting-start-with-celldepot.html#celldepot-api-application-programming-interface"><i class="fa fa-check"></i><b>2.5</b> CellDepot API (Application Programming Interface)</a></li>
<li class="chapter" data-level="2.6" data-path="getting-start-with-celldepot.html"><a href="getting-start-with-celldepot.html#code-availability"><i class="fa fa-check"></i><b>2.6</b> Code availability</a></li>
<li class="chapter" data-level="2.7" data-path="getting-start-with-celldepot.html"><a href="getting-start-with-celldepot.html#online-tutorials"><i class="fa fa-check"></i><b>2.7</b> Online tutorials</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="SITable.html"><a href="SITable.html"><i class="fa fa-check"></i><b>3</b> Supplemental Tables</a>
<ul>
<li><a href="SITable.html#table-s1---comparison-matrix-of-web-portal-tools"><span>Table S1</span> - Comparison matrix of web portal tools</a></li>
<li><a href="SITable.html#table-s2---criterion-for-query-search"><span>Table S2</span> - Criterion for query search</a></li>
<li><a href="SITable.html#table-s3---criterion-for-data-analysis-explorer"><span>Table S3</span> - Criterion for data analysis explorer</a></li>
<li><a href="SITable.html#table-s4---project-metadata-captured-in-celldepot"><span>Table S4</span> - Project metadata captured in CellDepot</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="SITutorial.html"><a href="SITutorial.html"><i class="fa fa-check"></i><b>4</b> Supplemental Tutorial</a>
<ul>
<li class="chapter" data-level="4.1" data-path="SITutorial.html"><a href="SITutorial.html#browse-projects"><i class="fa fa-check"></i><b>4.1</b> Browse Projects</a></li>
<li class="chapter" data-level="4.2" data-path="SITutorial.html"><a href="SITutorial.html#search"><i class="fa fa-check"></i><b>4.2</b> Search Projects</a></li>
<li class="chapter" data-level="4.3" data-path="SITutorial.html"><a href="SITutorial.html#project-filters"><i class="fa fa-check"></i><b>4.3</b> Project Filters</a></li>
<li class="chapter" data-level="4.4" data-path="SITutorial.html"><a href="SITutorial.html#visualize-datasets"><i class="fa fa-check"></i><b>4.4</b> Visualize Datasets</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="SITutorial.html"><a href="SITutorial.html#view-details"><i class="fa fa-check"></i><b>4.4.1</b> View Details</a></li>
<li class="chapter" data-level="4.4.2" data-path="SITutorial.html"><a href="SITutorial.html#figures6"><i class="fa fa-check"></i><b>4.4.2</b> Data Visualization and Analysis by cellxgene VIP</a></li>
<li class="chapter" data-level="4.4.3" data-path="SITutorial.html"><a href="SITutorial.html#case-study-1"><i class="fa fa-check"></i><b>4.4.3</b> Case Study 1</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="SITutorial.html"><a href="SITutorial.html#search-genes"><i class="fa fa-check"></i><b>4.5</b> Search Genes</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="SITutorial.html"><a href="SITutorial.html#case-study-2"><i class="fa fa-check"></i><b>4.5.1</b> Case Study 2</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="SITutorial.html"><a href="SITutorial.html#import"><i class="fa fa-check"></i><b>4.6</b> Import Projects</a></li>
<li class="chapter" data-level="4.7" data-path="SITutorial.html"><a href="SITutorial.html#create-project"><i class="fa fa-check"></i><b>4.7</b> Create Project</a></li>
<li class="chapter" data-level="4.8" data-path="SITutorial.html"><a href="SITutorial.html#update-a-project"><i class="fa fa-check"></i><b>4.8</b> Update a Project</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Supplementary Materials - CellDepot: A unified repository for scRNAseq data and visual exploration</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="SITutorial" class="section level1" number="4">
<h1><span class="header-section-number">Chapter 4</span> Supplemental Tutorial</h1>
<p>CellDepot is a scRNA-seq data portal consisting of a relational database management system, a graphical query builder, and data visualization tools, which can be accessed via the link, <a href="http://celldepot.bxgenomics.com" class="uri">http://celldepot.bxgenomics.com</a> for public datasets or a link to private installation, e.g., <a href="http://go.biogen.com/CellDepot" class="uri">http://go.biogen.com/CellDepot</a> for Biogen internal data collection. This is the supplemental tutorial providing detailed instructions. Clicking on a figure will bring up the enlarged view.</p>
<p><a href="figures/S1.jpg"><img src="figures/S1.jpg" alt="Figure S1" /></a></p>
<p>Figure S1. CellDepot Homepage. Functions designated to different user roles are highlighted in red callout boxes.</p>
<p>The interface contains multiple tabs at the top of the homepage, which correspond to major functionalities of CellDepot. All users can explore the existing datasets loaded in the public CellDepot for visualization and analysis while only admin users can upload datasets to a CellDepot instance, public or private.</p>
<div id="browse-projects" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Browse Projects</h2>
<p>Users can customize columns to be displayed by clicking on the green ‘Column Settings’ button. In addition, all or selected entries could be exported in CSV format. A quick search box is provided on the top-right corner of the table while building a complex query is exemplified in Figure S3 (<a href="SITutorial.html#search">4.2</a>).</p>
<p><a href="figures/S2.jpg"><img src="figures/S2.jpg" alt="Figure S2" /></a>
Figure S2. Browsing projects in a personalized view.</p>
</div>
<div id="search" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Search Projects</h2>
<p>This function allows users to search projects of interest, which can be accessed through the homepage as well. Users can search projects by 17 attributes in multiple logic conditions: annotation groups, cell count, cellxgene VIP launch method, Custom accession, description, DOI, file name, file size, gene count, name, notes, PMC ID, Publication Title, PubMed ID, Species, URL, and/or Year.</p>
<p><a href="figures/S3.jpg"><img src="figures/S3.jpg" alt="Figure S3" /></a>
Figure S3. Workflow of searching projects by using the graphical multi-logic, multi-condition query builder. Six datasets are identified when searching by ‘Species is Human’ and ‘Annotation Group contains Neuron.’</p>
</div>
<div id="project-filters" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Project Filters</h2>
<p>This function lists the filtered datasets simply based on AND logic operation of checked items under various categories. It is a friendly feature for first-time users as they may not be familiar with the design of the database to construct more complex query.</p>
<p><a href="figures/S4.jpg"><img src="figures/S4.jpg" alt="Figure S4" /></a>
Figure S4. The ‘Project Filters’ page. 29 records are identified by filtering criteria of ‘Year’ equaling 2021 and ‘Species’ being human.</p>
</div>
<div id="visualize-datasets" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Visualize Datasets</h2>
<div id="view-details" class="section level3" number="4.4.1">
<h3><span class="header-section-number">4.4.1</span> View Details</h3>
<p>The dataset information consists of project summary and annotation groups. The project summary is provided by admin users when uploading projects while annotation groups are retrieved from uploaded h5ad files.</p>
<p><a href="figures/S5.jpg"><img src="figures/S5.jpg" alt="Figure S5" /></a>
Figure S5. How to view details of a dataset. Steps are outlined by red circled numbers.</p>
</div>
<div id="figures6" class="section level3" number="4.4.2">
<h3><span class="header-section-number">4.4.2</span> Data Visualization and Analysis by cellxgene VIP</h3>
<p>CellDepot is not only a database management system, but also a web portal for visualizing and analyzing scRNA-seq datasets through embedded cellxgene VIP tool. By clicking ‘cellxgene VIP’ to access functional modules on the menu, users can perform advanced data visualization and analysis. To learn how to use cellxgene VIP, please go to <a href="https://interactivereport.github.io/cellxgene_VIP/tutorial/docs/how-to-use-cellxgene-vip.html" class="uri">https://interactivereport.github.io/cellxgene_VIP/tutorial/docs/how-to-use-cellxgene-vip.html</a>.</p>
<p><a href="figures/S6.jpg"><img src="figures/S6.jpg" alt="Figure S6" /></a>
Figure S6. Visualization and analysis of a scRNA-seq dataset by cellxgene VIP.</p>
</div>
<div id="case-study-1" class="section level3" number="4.4.3">
<h3><span class="header-section-number">4.4.3</span> Case Study 1</h3>
<p>Exploration and visualization of differentially expressed genes (DEGs) between two types of cells.</p>
<p>As shown in Figure S7a, two types of cells, Astrocytes (1036 cells) and Oligodendrocytes (4417 cells) are selected. By running differential gene expression analysis with one of the built-in statistical methods such as Welch’s t-test, we detected 1578 (DEGs), including 715 up-regulated and 853 down-regulated genes in astrocytes compared to oligodendrocytes (Figure S7a). The expression of the top four DEGs among the cell types indicates that gene MBP, ST18 and RNF220 are expressed explicitly in oligodendrocytes, while gene PITPNC3 is expressed mainly in astrocytes and endothelial cells (Figure S7b).</p>
<p><a href="figures/S7.jpg"><img src="figures/S7.jpg" alt="Figure S7" /></a>
Figure S7. Exploration of differentially expressed genes in dataset GSE140231 through cellxgene VIP. (a) Identifying differentially expressed genes in astrocytes and oligodendrocytes. (b) The expression of top four genes in various cell types as shown in dot plot.</p>
</div>
</div>
<div id="search-genes" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Search Genes</h2>
<p>This tab allows searching on genes of interest with the expression cutoff. The search outcome provides users a list of projects in which genes of interest are expressed above the cutoff. Each project displays a link to project page and a plot if applicable. This plot can be either a violin plot or dot plot showing the gene expression level in a selected annotation group. Further, under “Advanced Options,” users can define the range of expression color scale and the percentage represented by the largest dot to have expression data from various projects plotted in a unified manner.</p>
<p><a href="figures/S8.jpg"><img src="figures/S8.jpg" alt="Figure S8" /></a>
Figure S8. Steps to find gene expression level of a gene in projects under ‘Search Genes’ tab. The final plot can be customized by available options listed in red callout boxes on the left side.</p>
<div id="case-study-2" class="section level3" number="4.5.1">
<h3><span class="header-section-number">4.5.1</span> Case Study 2</h3>
<p>Cross-project comparison of skeletal muscle marker genes PAX3, PAX7, PITX2, MYF5, MYF6, MYOD1, MYOG, NEB, and MYH3 among the datasets whose species is human and cell type is myogenic.</p>
<p><a href="figures/s9.jpg"><img src="figures/S9.jpg" alt="Figure S9" /></a>
Figure S9. Workflow of conducting the cross-project comparison of a list of genes among the selected datasets.</p>
</div>
</div>
<div id="import" class="section level2" number="4.6">
<h2><span class="header-section-number">4.6</span> Import Projects</h2>
<p>The functionality is limited to admin users. To upload new projects to CellDepot database in batch, two types of files are required: 1) .h5ad files and 2) project information file in CSV (Comma Separated Values) format. First, the prepared h5ad files are required to be copied to a folder defined in the configuration file, e.g., /data/celldepot/all_h5ad_files/. Afterwards, admin users navigate to the CellDepot home page, click ‘Import Projects’ at the top menu, then ‘Download Example File’ to fill in meta information of datasets into the downloaded template for submission. In addition, there are two cellxgene VIP launch modes to chosen from, ‘Standard’ and ‘Preload in Memory.’ ‘Standard’ mode is for infrequently used datasets while ‘Preload in Memory’ should be selected to speed up loading and responding time of frequently used large datasets.
After the metadata file is uploaded, CellDepot will automatically convert the dataset to CSC format if needed through a cron job (<a href="getting-start-with-celldepot.html#cron">2.4</a>). To explore the detail of imported datasets, users can enter ‘Browse Projects’ page and then search these datasets by user assigned ‘Custom Accession’ identifiers.</p>
<p><a href="figures/S10.jpg"><img src="figures/S10.jpg" alt="Figure S10" /></a>
Figure S10. Workflow of how to import new datasets.</p>
</div>
<div id="create-project" class="section level2" number="4.7">
<h2><span class="header-section-number">4.7</span> Create Project</h2>
<p>Besides batch uploading under “Import Projects” tab, admin user can use the online form under this tab to submit information of a project.</p>
</div>
<div id="update-a-project" class="section level2" number="4.8">
<h2><span class="header-section-number">4.8</span> Update a Project</h2>
<p>Project information including launch mode can be modified by admin users by clicking on ‘Update’ link of a project under ‘Actions’ column in the table.</p>
<p><a href="figures/s11.jpg"><img src="figures/S11.jpg" alt="Figure S11" /></a></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="SITable.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/03-SITutorial.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["CellDepot.pdf", "CellDepot.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
},
"pandoc_args": ["--lua-filter=scholarly-metadata.lua", "--lua-filter=author-info-blocks.lua"]
});
});
</script>

</body>

</html>
