---
title: "Supplementary Materials - CellDepot: A unified repository for scRNAseq data and visual exploration "
author:
  - name: Dongdong Lin
    email: lindongdo@gmail.com
    affiliation: Biogen
  - name: Yirui Chen
    email: yirui.chen@biogen.com
    affiliation: Biogen
  - name: Soumya Negi
    email: soumya.negi@biogen.com
    affiliation: Biogen
  - name: Derrick Cheng
    email: derrick@bioinforx.com
    affiliation: BioInfoRx
  - name: Zhengyu Ouyang
    email: oyoung@bioinforx.com
    affiliation: BioInfoRx
  - name: David Sexton
    email: david.sexton@biogen.com
    affiliation: Biogen
  - name: Kejie Li
    email: kejie.li@biogen.com
    affiliation: Biogen
    footnote: Corresponding Author
  - name: Baohong Zhang
    email: baohong.zhang@biogen.com
    affiliation: Biogen
    footnote: Corresponding Author
address:
  - code: Biogen, Inc
    address: 225 Binney St, Cambridge, MA 02142, USA
  - code: BioInfoRx, Inc
    address: 510 Charmany Dr, Suite 275A, Madison, WI 53719, USA
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
classoption: openany
geometry: margin=2cm
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
github-repo: https://github.com/interactivereport/CellDepot
output: radix::radix_article
---
# Preface
This is a *supplementary materials* written in Markdown, which provides the detailed guide for CellDepot web portal.


```{r include=FALSE}
# automatically create a bib database for R packages
knitr::write_bib(c(
  .packages(), 'bookdown', 'knitr', 'rmarkdown'
), 'packages.bib')
```

<!--chapter:end:index.Rmd-->

# Getting start with CellDepot


## Sources of annotation and metadata
The original metadata information of each scRNA-seq dataset is retrieved from h5ad file, which is a preferred way of sharing and storing an on-disk representation of anndata object. When importing the dataset to the system, user inputs additional metadata information as shown in (\@ref(import)). Both metadata are collected and stored in a MySQL database table that is presented at http://celldepot.bxgenomics.com and Biogen internal instance, http://go.biogen.com/CellDepot.

## Data format, availability, and preparation

CellDepot requires scRNA-seq data in h5ad file where the expression matrix is stored in CSC (compressed sparse column) instead of CSR (compressed sparse row) format to improve the speed of data retrieving. For example, designating genes as columns in the h5ad file creates the interactive plot five times faster than as rows. Just in case, we provide sample scripts to help users generate h5ad files. Having gene expression matrix, metadata, and layout files, users can easily combine and convert their data to h5ad file by following this R script on https://github.com/interactivereport/CellDepot/blob/main/toH5ad.R. In the case of lacking layout file, users can also create h5ad file by following the Jupyter notebook https://github.com/interactivereport/CellDepot/blob/main/raw2h5ad.ipynb with custom python script tailored to their own data. Categorical features extracted from a h5ad file are shown in the 'annotation groups' column of the table on CellDepot home page, while the numerical features are shown as the histograms in the rightmost panel on cellxgene VIP. (\@ref(figures6))


## CellDepot platform and installation

The public version of CellDepot web portal is hosted at the web site,  http://celldepot.bxgenomics.com and Biogen internal link http://go.biogen.com/CellDepot. It is implemented with MySQL database, an advanced search engine, and powerful interactive visualizing tools that allow users to explore attributes of datasets as well as scRNA-seq analysis results. Also, users can intentionally select single-cell RNA-seq datasets on the web interface by simply browsing the online dataset table or applying advanced search to perform the cross-dataset comparison. Moreover, CellDepot also provides comprehensive data analysis tools via an embedded interactive visualization plugin. To host private datasets, local instance of CellDepot on Unix server can be installed by following the guide here, https://celldepot.bxgenomics.com/celldepot_manual/install_environment.php.

## How to set up cron job? {#cron}
The following cron job entry is needed to convert h5ad file to CSC format on the background,
```{bash, eval=FALSE}
@hourly <user-name> cd /var/www/html/celldepot/app/core; php ./api_toCSCh5ad.php
```
Note: Please make sure that the user has the permission to write in the data directory.

## CellDepot API (Application Programming Interface)
The CellDepot API web service provides a direct way to generate figures for users to share or embed in web page. For example, the following URL will generate a gene expression violin plot across cell clusters for IRAK4 gene for the data set with ID equaling one, https://celldepot.bxgenomics.com/celldepot/app/core/api_gene_plot.php?ID=1&Genes=IRAK4&Plot_Type=violin&Subsampling=0&n=0&g=0&Project_Group=CLUSTER. The complete format of the URL and explanation of parameters are detailed in the online documentation, https://celldepot.bxgenomics.com/celldepot_manual/api_gene_plot.php. 

## Code availability
The source code, links to tutorials and other supplementary documents are provided at https://github.com/interactivereport/CellDepot. With broad adoption and contribution in mind, CellDepot is released under the MIT open-source license. The detailed instruction of local installation is available at  https://celldepot.bxgenomics.com/celldepot_manual.

## Online tutorials
To better assist biologists to use CellDepot and integrated cellxgene VIP visual analytical tool, we created online easy-to-access HTML tutorials with step-by-step guides available at https://interactivereport.github.io/CellDepot/bookdown/docs/SITutorial.html for CellDepot and https://interactivereport.github.io/cellxgene_VIP/tutorial/docs/how-to-use-cellxgene-vip.html for cellxgene VIP, respectively. In addition, a question mark next to the title of each VIP function module provides a direct way to reach the corresponding section of the HTML document for help.


<!--chapter:end:04-Method.Rmd-->

---
output:
  pdf_document: default
  html_document: default
---
# Supplemental Tables {#SITable}

## [Table S1](https://github.com/interactivereport/CellDepot/blob/gh-pages/bookdown/S1.csv) - Comparison matrix of web portal tools {-}
Table link: https://github.com/interactivereport/CellDepot/blob/gh-pages/bookdown/S1.csv
```{r, echo=FALSE}
library(kableExtra)
df <- read.csv("S1.csv")
df %>%
  kbl() %>%
  kable_classic(full_width = F, html_font = "Cambria") %>%
  scroll_box(width = "100%")
```
Note: The criteria for query search and data analysis explorer please see Table S2 and S3.

#### Table S1 in one picture (click on the picture to zoom in) {-}
[![Table table_s1](https://interactivereport.github.io/CellDepot/bookdown/figures/table_s1.jpg)](https://interactivereport.github.io/CellDepot/bookdown/figures/table_s1.jpg)

## [Table S2](https://github.com/interactivereport/CellDepot/blob/gh-pages/bookdown/S2.csv) - Criterion for query search {-}
Table link: https://github.com/interactivereport/CellDepot/blob/gh-pages/bookdown/S2.csv
```{r, echo=FALSE}
library(kableExtra)
df <- read.csv("S2.csv")
df %>%
  kbl() %>%
  kable_classic(full_width = F, html_font = "Cambria") %>%
  scroll_box(width = "100%")
```

## [Table S3](https://github.com/interactivereport/CellDepot/blob/gh-pages/bookdown/S3.csv) - Criterion for data analysis explorer {-}
Table link: https://github.com/interactivereport/CellDepot/blob/gh-pages/bookdown/S3.csv
```{r, echo=FALSE}
library(kableExtra)
df <- read.csv("S3.csv")
df %>%
  kbl() %>%
  kable_classic(full_width = F, html_font = "Cambria") %>%
  scroll_box(width = "100%")
```

## [Table S4](https://github.com/interactivereport/CellDepot/blob/gh-pages/bookdown/S4.csv) - Project metadata captured in CellDepot {-}
Table link: https://github.com/interactivereport/CellDepot/blob/gh-pages/bookdown/S4.csv
```{r, echo=FALSE}
library(kableExtra)
df <- read.csv("S4.csv")
df %>%
  kbl() %>%
  kable_classic(full_width = F, html_font = "Cambria") %>%
  scroll_box(width = "100%")
```


<!--chapter:end:01-SITable.Rmd-->

# Supplemental Tutorial {#SITutorial}

CellDepot is a scRNA-seq data portal consisting of a relational database management system, a graphical query builder, and data visualization tools, which can be accessed via the link, http://celldepot.bxgenomics.com for public datasets or a link to private installation, e.g., http://go.biogen.com/CellDepot for Biogen internal data collection. This is the supplemental tutorial providing detailed instructions. Clicking on a figure will bring up the enlarged view.


[![Figure S1](https://interactivereport.github.io/CellDepot/bookdown/figures/S1.jpg)](https://interactivereport.github.io/CellDepot/bookdown/figures/S1.jpg)

Figure S1. CellDepot Homepage. Functions designated to different user roles are highlighted in red callout boxes.

The interface contains multiple tabs at the top of the homepage, which correspond to major functionalities of CellDepot. All users can explore the existing datasets loaded in the public CellDepot for visualization and analysis while only admin users can upload datasets to a CellDepot instance, public or private.

## 	Browse Projects
Users can customize columns to be displayed by clicking on the green 'Column Settings' button. In addition, all or selected entries could be exported in CSV format. A quick search box is provided on the top-right corner of the table while building a complex query is exemplified in Figure S3 (\@ref(search)).

[![Figure S2](https://interactivereport.github.io/CellDepot/bookdown/figures/S2.jpg)](https://interactivereport.github.io/CellDepot/bookdown/figures/S2.jpg)
Figure S2. Browsing projects in a personalized view.


## Search Projects {#search}
This function allows users to search projects of interest, which can be accessed through the homepage as well. Users can search projects by 17 attributes in multiple logic conditions: annotation groups, cell count, cellxgene VIP launch method, Custom accession, description, DOI, file name, file size, gene count, name, notes, PMC ID, Publication Title, PubMed ID, Species, URL, and/or Year.

[![Figure S3](https://interactivereport.github.io/CellDepot/bookdown/figures/S3.jpg)](https://interactivereport.github.io/CellDepot/bookdown/figures/S3.jpg)
Figure S3. Workflow of searching projects by using the graphical multi-logic, multi-condition query builder. Six datasets are identified when searching by 'Species is Human' and 'Annotation Group contains Neuron'.


## Project Filters

This function lists the filtered datasets simply based on AND logic operation of checked items under various categories. It is a friendly feature for first-time users as they may not be familiar with the design of the database to construct more complex query.

[![Figure S4](https://interactivereport.github.io/CellDepot/bookdown/figures/S4.jpg)](https://interactivereport.github.io/CellDepot/bookdown/figures/S4.jpg)
Figure S4. The 'Project Filters' page. 29 records are identified by filtering criteria of 'Year' equaling 2021 and 'Species' being human.

## Visualize Datasets
### View Details

The dataset information consists of project summary and annotation groups. The project summary is provided by admin users when uploading projects while annotation groups are retrieved from uploaded h5ad files.

[![Figure S5](https://interactivereport.github.io/CellDepot/bookdown/figures/S5.jpg)](https://interactivereport.github.io/CellDepot/bookdown/figures/S5.jpg)
Figure S5. How to view details of a dataset. Steps are outlined by red circled numbers.


### Data Visualization and Analysis by cellxgene VIP {#figures6}
CellDepot is not only a database management system, but also a web portal for visualizing and analyzing scRNA-seq datasets through embedded cellxgene VIP tool. By clicking 'cellxgene VIP' to access functional modules on the menu, users can perform advanced data visualization and analysis. To learn how to use cellxgene VIP, please go to https://interactivereport.github.io/cellxgene_VIP/tutorial/docs/how-to-use-cellxgene-vip.html.

[![Figure S6](https://interactivereport.github.io/CellDepot/bookdown/figures/S6.jpg)](https://interactivereport.github.io/CellDepot/bookdown/figures/S6.jpg)
Figure S6. Visualization and analysis of a scRNA-seq dataset by cellxgene VIP.

### Case Study 1

Exploration and visualization of differentially expressed genes (DEGs) between two types of cells.

As shown in Figure S7a, two types of cells, Astrocytes (1036 cells) and Oligodendrocytes (4417 cells) are selected. By running differential gene expression analysis with one of the built-in statistical methods such as Welch's t-test, we detected 1578 (DEGs), including 715 up-regulated and 853 down-regulated genes in astrocytes compared to oligodendrocytes (Figure S7a). The expression of the top four DEGs among the cell types indicates that gene MBP, ST18 and RNF220 are expressed explicitly in oligodendrocytes, while gene PITPNC3 is expressed mainly in astrocytes and endothelial cells (Figure S7b).

[![Figure S7](https://interactivereport.github.io/CellDepot/bookdown/figures/S7.jpg)](https://interactivereport.github.io/CellDepot/bookdown/figures/S7.jpg)
Figure S7. Exploration of differentially expressed genes in dataset GSE140231 through cellxgene VIP. (a) Identifying differentially expressed genes in astrocytes and oligodendrocytes. (b) The expression of top four genes in various cell types as shown in dot plot.

## Search Genes

This tab allows searching on genes of interest with the expression cutoff. The search outcome provides users a list of projects in which genes of interest are expressed above the cutoff.  Each project displays a link to project page and a plot if applicable. This plot can be either a violin plot or dot plot showing the gene expression level in a selected annotation group. Further, under "Advanced Options", users can define the range of expression color scale and the percentage represented by the largest dot to have expression data from various projects plotted in a unified manner. 

[![Figure S8](https://interactivereport.github.io/CellDepot/bookdown/figures/S8.jpg)](https://interactivereport.github.io/CellDepot/bookdown/figures/S8.jpg)
Figure S8. Steps to find gene expression level of a gene in projects under 'Search Genes' tab. The final plot can be customized by available options listed in red callout boxes on the left side.

### Case Study 2

Cross-project comparison of skeletal muscle marker genes PAX3, PAX7, PITX2, MYF5, MYF6, MYOD1, MYOG, NEB, and MYH3 among the datasets whose species is human and cell type is myogenic.  

[![Figure S9](https://interactivereport.github.io/CellDepot/bookdown/figures/S9.jpg)](https://interactivereport.github.io/CellDepot/bookdown/figures/S9.jpg)
Figure S9. Workflow of conducting the cross-project comparison of a list of genes among the selected datasets.


## Import Projects {#import}
The functionality is limited to admin users. To upload new projects to CellDepot database in batch, two types of files are required: 1) .h5ad files and 2) project information file in CSV (Comma Separated Values) format. First, the prepared h5ad files are required to be copied to a folder defined in the configuration file, e.g., /data/celldepot/all_h5ad_files/. Afterwards, admin users navigate to the CellDepot home page, click ‘Import Projects’ at the top menu, then ‘Download Example File’ to fill in meta information of datasets into the downloaded template for submission. In addition, there are two cellxgene VIP launch modes to chosen from, 'Standard' and 'Preload in Memory'. 'Standard' mode is for infrequently used datasets while 'Preload in Memory' should be selected to speed up loading and responding time of frequently used large datasets.
After the metadata file is uploaded, CellDepot will automatically convert the dataset to CSC format if needed through a cron job (\@ref(cron)). To explore the detail of imported datasets, users can enter ‘Browse Projects’ page and then search these datasets by user assigned 'Custom Accession' identifiers.

[![Figure S10](https://interactivereport.github.io/CellDepot/bookdown/figures/S10.jpg)](https://interactivereport.github.io/CellDepot/bookdown/figures/S10.jpg)
Figure S10. Workflow of how to import new datasets.

## Create Project
Besides batch uploading under "Import Projects" tab, admin user can use the online form under this tab to submit information of a project.

## Update a Project
Project information including launch mode can be modified by admin users by clicking on 'Update' link of a project under 'Actions' column in the table.

[![Figure S11](https://interactivereport.github.io/CellDepot/bookdown/figures/S11.jpg)](https://interactivereport.github.io/CellDepot/bookdown/figures/S11.jpg)

<!--chapter:end:03-SITutorial.Rmd-->

